Extensions/Guardrail+Settings.swift

```diff
@@ -84,58 +84,74 @@ public extension Guardrail where Value == HKQuantity {
         recommendedBounds: 16...399,
         unit: HKUnit.milligramsPerDeciliter.unitDivided(by: .internationalUnit())
     )
-
+ 
+    static let absoluteMinimumCarbRatio = 2.0
     static let carbRatio = Guardrail(
-        absoluteBounds: 1...150,
-        recommendedBounds: 3.0.nextUp...28.0.nextDown,
+        absoluteBounds: absoluteMinimumCarbRatio...150,
+        recommendedBounds: 4...28,
         unit: .gramsPerUnit
     )
 
     static func basalRate(supportedBasalRates: [Double]) -> Guardrail {
+        let scheduledBasalRateAbsoluteRange = 0.05...30.0
         let recommendedLowerBound = supportedBasalRates.first == 0
             ? supportedBasalRates.dropFirst().first!
             : supportedBasalRates.first!
         return Guardrail(
-            absoluteBounds: supportedBasalRates.first!...supportedBasalRates.last!,
-            recommendedBounds: recommendedLowerBound...supportedBasalRates.last!,
+            absoluteBounds: (supportedBasalRates.first!...supportedBasalRates.last!).clamped(to: scheduledBasalRateAbsoluteRange),
```

Comment:
[nit] This will result in an `absoluteBounds` and `recommendedBounds` whose end points are not necessarily supported basal rates. Is this intended?

For example, if the supported basal rates were `[0.03, 0.06, ... 32.7, 33.0]` (with increments of 0.3) then `absoluteBounds` would be clamped on both ends by `scheduledBasalRateAbsoluteRange` and result in `0.05...30.0`, but `0.05` is not a supported basal rate. 

Are you sure you don't want to find the lowest and highest supported basal rate than is within the `scheduledBasalRateAbsoluteRange`?

Reply:
Good point.  I am just coming up to speed on how this all works.  The **Done** criteria were not stated in this way, and I'd assumed that this would be correct.  I'll look into it.